src/EditorFeatures/Test/CodeGeneration/CodeGenerationTests.cs

```diff
@@ -354,11 +354,13 @@ public partial class CodeGenerationTests
             }
 
             using var context = await TestContext.CreateAsync(initial, expected);
+            var workspace = context.Workspace;
             if (options != null)
             {
                 foreach (var kvp in options)
                 {
-                    context.Workspace.Options = context.Workspace.Options.WithChangedOption(kvp.Key, kvp.Value);
+                    workspace.TryApplyChanges(workspace.CurrentSolution.WithOptions(workspace.Options
+                        .WithChangedOption(kvp.Key, kvp.Value)));
                 }
```

Comment:
💡 Update the options set in a loop, then call `TryApplyChanges` once with the result